.github/workflows/on_release_tag.yml

```diff
@@ -0,0 +1,127 @@
+name: On release tag
+
+on:
+  push:
+    tags:
+      - '^[0-9]+\.[0-9]+\.[0-9]+(?!.*-SNAPSHOT).*$'
+
+jobs:
+  detekt:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: gradle/wrapper-validation-action@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@417ae3ccd767c252f5661f1ace9f835f9654f2b5 # v3.1.0
+      - name: Run Detekt on commonMain
+        run: ./gradlew detektCommonMain
+
+  validate-binary-compatibility:
+    runs-on: macos-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: gradle/wrapper-validation-action@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@417ae3ccd767c252f5661f1ace9f835f9654f2b5 # v3.1.0
+      - name: Validate binary compatibility
+        run: ./gradlew apiCheck
+
+  test-public-api:
+    runs-on: macos-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+      - uses: gradle/wrapper-validation-action@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@417ae3ccd767c252f5661f1ace9f835f9654f2b5 # v3.1.0
+      - name: Test public API
+        run: ./gradlew :apiTester:assemble
+
+  build:
+    needs: [ detekt, validate-binary-compatibility, test-public-api ]
+    runs-on: macos-latest
+    permissions:
+      contents: read
+    steps:
+    - uses: actions/checkout@v4
+    - uses: gradle/wrapper-validation-action@v2
+    - name: Set up JDK 17
+      uses: actions/setup-java@v4
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+
+    # Configure Gradle for optimal use in GitHub Actions, including caching of downloaded dependencies.
+    # See: https://github.com/gradle/actions/blob/main/setup-gradle/README.md
+    - name: Setup Gradle
+      uses: gradle/actions/setup-gradle@417ae3ccd767c252f5661f1ace9f835f9654f2b5 # v3.1.0
+
+    - name: Build libraries
+      run: ./gradlew assemble -x :composeApp:assemble
+    - name: Build sample
+      run: ./gradlew :composeApp:assembleDebug
+
+  generate-api-reference:
+    needs: build
```

Comment:
Hmm since everything before this is shared with the `main` workflow, is there a way to extract it so we avoid the duplication?

Reply:
Totally agree, I think we can do something with [composite actions](https://docs.github.com/en/actions/using-workflows/avoiding-duplication). But I'm not sure how much we'll be using GHA, especially the `main` workflow, long term. So I'd propose to wait and see how that develops and work on organizing the workflows at that point.

Reply:
Makes sense to me 👍 Happy to merge this!